### DISCUSSION TITLE
vanna and watsonx.ai integration

### DISCUSSION THREAD

**Discussion by arabenandrasana** [vanna-ai/vanna#633] [Q&A]
Does Vanna integration with watsonx.ai (calling LLM through watsonx.ai APIs) planned or can be implemented?
Thank you

**Comment by zainhoda** [comment]
We've considered it but please tell us more! Why are you using watsonx? You're the first person to request this. If you prefer to e-mail us we can discuss over e-mail as well.

**Reply by arabenandrasana** [reply]
Because I'm actually working for IBM with some watsonx customers and I was wondering if on paper it's something doable either by Vanna or by a system integrator if needed.

**Reply by zainhoda** [reply]
@arabenandrasana yes -- Vanna can work with any LLM and any vector database. Just follow the documentation for "Other LLM" and "Other Vector Database" 

Once you implement the necessary abstract methods, please feel free to submit a PR like Qdrant did here:

https://github.com/vanna-ai/vanna/pull/362
